sherpa/utils/testing.py

```diff
@@ -0,0 +1,279 @@
+           The expected value, or values.
+        second : number or array_like
+           The value, or values, to check. If first is an array, then
+           second must be an array of the same size. If first is
+           a scalar then second can be a scalar or an array.
+        tol : number
+           The absolute tolerance used for comparison.
+        msg : string
+           The message to display if the check fails.
+
+        """
+
+        self.assertFalse(numpy.any(sao_fcmp(first, second, tol)), msg)
+
+    def assertNotEqualWithinTol(self, first, second, tol=1e-7, msg=None):
+        """Check that the values are not equal within an absolute tolerance.
+
+        Parameters
+        ----------
+        first : number or array_like
+           The expected value, or values.
+        second : number or array_like
+           The value, or values, to check. If first is an array, then
+           second must be an array of the same size. If first is
+           a scalar then second can be a scalar or an array.
+        tol : number
+           The absolute tolerance used for comparison.
+        msg : string
+           The message to display if the check fails.
+
+        """
+
+        self.assertTrue(numpy.all(sao_fcmp(first, second, tol)), msg)
+
+    # for running regression tests from sherpa-test-data
+    def run_thread(self, name, scriptname='fit.py'):
+        """Run a regression test from the sherpa-test-data submodule.
+
+        Parameters
+        ----------
+        name : string
+           The name of the science thread to run (e.g., pha_read,
+           radpro). The name should match the corresponding thread
+           name in the sherpa-test-data submodule. See examples below.
+        scriptname : string
+           The suffix of the test script file name, usually "fit.py."
+
+        Examples
+        --------
+        Regression test script file names have the structure
+        "name-scriptname.py." By default, scriptname is set to "fit.py."
+        For example, if one wants to run the regression test
+        "pha_read-fit.py," they would write
+
+        >>> run_thread("pha_read")
+
+        If the regression test name is "lev3fft-bar.py," they would do
+
+        >>> run_thread("lev3fft", scriptname="bar.py")
+
+        """
+
+        scriptname = name + "-" + scriptname
+        self.locals = {}
+        cwd = os.getcwd()
+        os.chdir(self.datadir)
+        try:
+            with open(scriptname, "rb") as fh:
+                cts = fh.read()
+            exec(compile(cts, scriptname, 'exec'), {}, self.locals)
+        finally:
+            os.chdir(cwd)
+
+
+def has_package_from_list(*packages):
+    """
+    Returns True if at least one of the ``packages`` args is importable.
+    """
+    for package in packages:
+        try:
+            importlib.import_module(package)
+            return True
+        except:
+            pass
+    return False
+
+
+def requires_package(msg=None, *packages):
```

Comment:
I guess it comes down to whether we think of `requires_package` being an internal routine to this module or if there's ever chance it could be used directly in a test. I would prefer the second one but haven't really thought about it that deeply.